src/read.jl

```diff
@@ -88,117 +84,108 @@ function read_header(io::IO)
     # we ignore the above entirely and use `recording_id.startdate`. We could add a
     # check here on `year(today())`, but that will be dead code for the next 60+ years.
 
-    nb_header = parse(Int, String(Base.read(io, 8)))
+    header_byte_count = parse(Int, String(Base.read(io, 8)))
     reserved = String(Base.read(io, 44))
     continuous = !startswith(reserved, "EDF+D")
     n_records = parse(Int, String(Base.read(io, 8)))
     duration = parse(Float64, String(Base.read(io, 8)))
-    n_signals = parse(Int, String(Base.read(io, 4)))
-
-    signals = [Signal() for _ = 1:n_signals]
+    signal_count = parse(Int, String(Base.read(io, 4)))
 
-    # TODO: EDF+ allows floating point data which does not fit within the Int16 limits.
-    # See https://edfplus.info/specs/edffloat.html for details. MNE seems to implement
-    # this(?)
-
-    set!(io, signals, :label, 16)
-    set!(io, signals, :transducer, 80)
-    set!(io, signals, :physical_units, 8)
-    set!(io, parse_float, signals, :physical_min, 8)
-    set!(io, parse_float, signals, :physical_max, 8)
-    set!(io, parse_float, signals, :digital_min, 8)
-    set!(io, parse_float, signals, :digital_max, 8)
-    set!(io, signals, :prefilter, 80)
-    set!(io, x->parse(Int16, x), signals, :n_samples, 8)
-
-    skip(io, 32 * n_signals)  # Reserved
+    header = FileHeader(version, patient_id, recording_id, start,
+                        continuous, n_records, duration)
+    return header, header_byte_count, signal_count
+end
 
-    for signal in signals
-        signal.samples = Vector{Int16}()  # Otherwise it's #undef
-    end
+const FIELD_SIZES = [16, 80, 8, 8, 8, 8, 8, 80, 8]
+const PARSE_METHODS = Function[strip, strip, strip, parse_float, parse_float, parse_float, parse_float, strip, x -> parse(Int16, x)]
 
-    anno_idx = findfirst(signal->signal.label == "EDF Annotations", signals)
-    if anno_idx !== nothing
-        n_signals -= 1
+function read_field(io::IO, f::Function, signal_count::Integer, size::Integer)
+    fields = []
+    for i in 1:signal_count
+        push!(fields, f(String(Base.read(io, size))))
     end
-
-    @assert position(io) == nb_header
-
-    h = Header(version, patient_id, recording_id, continuous, start, n_records,
-               duration, n_signals)
-    return (h, signals, anno_idx)
+    return fields
 end
 
-function read_data!(io::IO, signals::Vector{Signal}, header::Header, ::Nothing)
-    for i = 1:header.n_records, signal in signals
-        append!(signal.samples, reinterpret(Int16, Base.read(io, 2 * signal.n_samples)))
+function read_signals(io::IO, file_header::FileHeader, signal_headers::Vector)
+    annotation_header = findfirst(header -> header.label == "EDF Annotations", signal_headers)
+    if annotation_header !== nothing
+        signal_headers[annotation_header] = EDF.AnnotationListHeader(signal_headers[annotation_header])
```

Comment:
```suggestion
        signal_headers[annotation_header] = AnnotationListHeader(signal_headers[annotation_header])
```
It's not necessary to qualify names with the current module name within that module.